components/table/Table.tsx

```diff
@@ -50,8 +50,8 @@ export type TableColumnConfig<T> = ColumnProps<T>;
 
 export interface TableRowSelection<T> {
   type?: 'checkbox' | 'radio';
-  selectedRowKeys?: string[];
-  onChange?: (selectedRowKeys: string[], selectedRows: Object[]) => any;
+  selectedRowKeys?: string[] | number[];
```

Comment:
React allows number type keys (see https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L31). I am aware these are unofficial but are the most popular. I have been using number type keys with antd Table in JavaScript which worked fine, I have just migrated our codebase to TypeScript which also works but I can only compile if I tweak these interfaces to allow number[] as well.